### ISSUE TITLE
Inconsistent detection of hardcoded paths with/without ripgrep

### ISSUE THREAD

**Issue by jaimergp** [conda/conda-build#3737]
### Actual Behavior

At least in Windows, when `ripgrep` is present (`rg` in `PATH`), `conda-build` is unable to find a `*.py` file that contains a hardcoded `PREFIX` that should be replaced. If `ripgrep` is removed, then `conda build` does find it thanks to the [fallback](https://github.com/conda/conda-build/blob/master/conda_build/build.py#L186-L216) (slow) mechanism.

### Expected Behavior

The file should be found with and without `ripgrep`.

### Steps to Reproduce

I have seen this when pushing a recipe to `conda-forge` (this the [PR](https://github.com/conda-forge/staged-recipes/pull/9163)), and then reproduced locally in Windows.

1. Get the [recipe](https://github.com/jaimergp/staged-recipes/tree/openmm-cpu/recipes/openmm-cpu).
2. `conda build` it
3. The tests will pass if the hardcoded path at `$PREFIX/Lib/site-packages/simtk/openmm/version.py` is found and replaced. This will depend on `rg` being present or not.

Demo contents of `version.py`:

```
# THIS FILE IS GENERATED FROM OPENMM SETUP.PY
short_version = '7.4.0'
version = '7.4.0'
full_version = '7.4.0.dev-Unknown'
git_revision = 'Unknown'
release = False
openmm_library_path = r'C:/bld/openmm-cpu_1568802222237/_h_env/Library/lib'
if not release:
    version = full_version
```

Check different behavior in Azure vs Appveyor:

* [Logs for Azure](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=73158&view=logs&jobId=240f1fee-52bc-5498-a14a-8361bde76ba0&taskId=23010c01-6a9d-5f94-9037-792618644b77&lineStart=4807&lineEnd=4808&colStart=1&colEnd=1) (ripgrep message is printed, which means it's not being used, hardcoded path is found)
* [Logs for Appveyor](https://ci.appveyor.com/project/conda-forge/staged-recipes/builds/27491246#L4915) (no ripgrep message is printed, rg is being used, the hardcoded path is not found, tests fail)

##### `Output of conda info`

```
     active environment : base
    active env location : C:\Users\Jaime\Miniconda3
            shell level : 1
       user config file : C:\Users\Jaime\.condarc
 populated config files : C:\Users\Jaime\.condarc
          conda version : 4.7.11
    conda-build version : 3.18.9
         python version : 3.6.7.final.0
       virtual packages :
       base environment : C:\Users\Jaime\Miniconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\Jaime\Miniconda3\pkgs
                          C:\Users\Jaime\.conda\pkgs
                          C:\Users\Jaime\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\Jaime\Miniconda3\envs
                          C:\Users\Jaime\.conda\envs
                          C:\Users\Jaime\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.7.11 requests/2.22.0 CPython/3.6.7 Windows/10 Windows/10.0.18980
          administrator : False
             netrc file : None
           offline mode : False
```

**Comment by y2kbugger** [comment]
This should be higher prio. Been thinking for a month that my problems with entrypoints were do to a change in the way Anaconda no longer adds the magic dll finder patch to python...

**Comment by gabrielcnr** [comment]
I came across the same problem today at work.
On Windows, if I have `ripgrep` installed, `conda-build` does not find the files that need the prefix relocation correctly and them to the `has_prefix` file that is bundled together with the conda package.

There I'm using `conda-build==3.18.6`:
https://github.com/conda/conda-build/blob/3.18.6/conda_build/build.py#L191

However, I can see that in `master/HEAD` that part of the code is already receiving special treatment for Windows, so maybe this problem is correct in the latest release (TBC).
https://github.com/conda/conda-build/blob/master/conda_build/build.py#L204-L205

**Comment by marcelotrevisani** [comment]
It seems to be a legit problem. I will mark @mingwandroid who may know better about it

**Comment by gabrielcnr** [comment]
I just tested here with `conda-build==3.19.2` and `ripgrep==12.1.1` on Windows, and it built successfully a package with entry points with prefix inside `Scripts/`.
I think this issue can be closed.

**Comment by y2kbugger** [comment]
It might be the installing, not the building.

**Comment by gabrielcnr** [comment]
It's the build. I tested/verified by opening the conda package (`.tar.bz2` file) and then checking the contents of the `has_prefix` file there. Previously, on `3.18.*`, when building on Windows with `ripgrep` installed on the same environment as `conda-build`, that file was returning empty. Kicking off the build again from an environment without `ripgrep`, the file was populated correctly with the `*-script.py` files which contain prefix.
I tested today on `3.19.2` and the file was populated correctly when using `ripgrep`.
